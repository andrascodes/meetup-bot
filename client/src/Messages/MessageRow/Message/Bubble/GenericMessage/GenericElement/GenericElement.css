.GenericElement {
  background-color: #fff;
  color: #000;
  border-radius: 1em;
  border: 0.2em solid rgba(0, 0, 0, .10);
}

.GenericElementImage {
  border-radius: 1em 1em 0em 0em;
}

.GenericElementTitleContainer {
  padding: 0.4em;
}

.GenericElementTitle {
  color: #000;
  font-weight: bold;
}

.GenericElementSubtitle {
  color: #999;
}

.GenericElementButton {
  padding: 0.3em 0em;
  color: #0084ff;
  font-weight: 500;
}

.GenericDataContainer {
  border-bottom: 0.1em solid rgba(0, 0, 0, .10);
}