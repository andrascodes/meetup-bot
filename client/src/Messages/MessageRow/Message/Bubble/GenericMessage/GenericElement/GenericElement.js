import React from 'react'

import './GenericElement.css'

// image_url, title, subtitle, default_action.url, buttons, btn.title
export const GenericElement = props => {
  return (
    <div className="GenericElement">
      <img className="GenericElementImage img-responsive" src={props.element.image_url} alt=""/>
      <div className="GenericElementTitleContainer GenericDataContainer text-left">
        <div className="GenericElementTitle">
          {props.element.title}
        </div>
        <div className="GenericElementSubtitle">
          {props.element.subtitle}
        </div>
        <div className="GenericElementDefaultAction GenericElementSubtitle">
          {props.element.default_action && new URL(props.element.default_action.url).hostname}
        </div>
      </div>
      {props.element.buttons.map((btn, ind) => (
        <div className="GenericElementButton GenericDataContainer text-center" key={ind}>
          {btn.title}
        </div>
      ))}
    </div>
  )
}