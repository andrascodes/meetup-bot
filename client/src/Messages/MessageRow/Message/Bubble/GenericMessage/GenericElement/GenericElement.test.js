import React from 'react'
import renderer from 'react-test-renderer'

import { GenericElement } from './GenericElement'

describe.only('GenericElement', () => {

  test('renders correctly', () => {
    const tree = renderer.create(
      <GenericElement
        element={{
          image_url: 'https://www.google.com',
          title: 'Title',
          subtitle: 'Subtitle',
          default_action: {
            url: 'https://www.google.com'
          },
          buttons: [
            {
              title: 'Btn title1',
            },
            {
              title: 'Btn title2',
            },
          ]
        }}
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

})