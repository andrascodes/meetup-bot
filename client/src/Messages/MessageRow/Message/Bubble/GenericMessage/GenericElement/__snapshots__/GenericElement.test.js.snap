exports[`GenericElement renders correctly 1`] = `
<div
  className="GenericElement">
  <img
    alt=""
    className="GenericElementImage img-responsive"
    src="https://www.google.com" />
  <div
    className="GenericElementTitleContainer GenericDataContainer text-left">
    <div
      className="GenericElementTitle">
      Title
    </div>
    <div
      className="GenericElementSubtitle">
      Subtitle
    </div>
    <div
      className="GenericElementDefaultAction GenericElementSubtitle">
      www.google.com
    </div>
  </div>
  <div
    className="GenericElementButton GenericDataContainer text-center">
    Btn title1
  </div>
  <div
    className="GenericElementButton GenericDataContainer text-center">
    Btn title2
  </div>
</div>
`;
